src/menelaus/concept_drift/md3.py

```diff
@@ -0,0 +1,358 @@
+    Support Vector Machine (SVM) because SVMs have explicit margins defined
+    by the hyperplane. Classifiers such as decision trees and K-nearest
+    neighbors return discrete class labels and do not have any intuitive
+    notion of margin. To implement compatibility with MD3 for these
+    classifiers, a feater bagging ensemble technique must be used (this will
+    be added to the MD3 implementation later on). For now, this algorithm
+    is designed specifically for SVMs (and potentially logistic regression) 
+    for the sake of simplicity.
+
+    Attributes:
+        total_updates (int): number of samples the drift detector has ever
+            been updated with
+        updates_since_reset (int): number of samples since the last time the
+            drift detector was reset
+        drift_state (str): detector's current drift state. Can take values
+            ``"drift"``, ``"warning"``, or ``None``.
+    """
+
+    input_type = "stream"
+
+    def __init__(self, clf, sensitivity=2, k=10, oracle_data_length_required=None):
+        """
+        Args:
+            clf (sklearn.svm.SVC): the classifier for which we are tracking drift.
+                For now, assumed to be a Support Vector Machine (SVM). TODO: change
+                this after adding capability for other models.
+            sensitivity (float): the sensitivity at which a change in margin density
+                will be detected. Change is signaled when the margin density at a
+                time t, given by MD sub t, deviates by more than sensitivity 
+                standard deviations from the reference margin density value MD sub Ref.
+                A larger value can be set if frequent signaling is not desired, 
+                alternatively a lower value could be used for critical applications
+                where small changes could be harmful, if undetected. Defaults to 2. 
+                TODO: suggested by paper to be picked in the range of [0, 3]
+            k (int): the number of folds that will be used in k-fold cross validation
+                when measuring the distribution statistics of the reference batch
+                of data. Defaults to 10.
+            oracle_data_length_required (int): the number of samples that will need to
+                be collected by the oracle when drift is suspected, for the purpose of
+                either confirming or ruling out drift, and then retraining the classifier
+                if drift is confirmed. Defaults to the length of the reference distribution
+                (this is set in the set_reference method).
+        """
+
+        super().__init__()
+        self.classifier = clf
+        self.sensitivity = sensitivity
+        self.k = k
+        self.oracle_data_length_required = oracle_data_length_required
+        self.oracle_data = None
+        self.waiting_for_oracle = False
+        self.process_svm()
+
+    def process_svm(self):
```

Comment:
1. Got rid of `calculate_margin_inclusion_signal_other_classifier`. Added an arg to `calculate_margin_inclusion_signal` so that it now takes the classifier, so it works for either case (taking the classifier or the duplicate classifier).
2. Created a `self.margin_calculation_function` in the `init`, which defaults to the `calculate_margin_inclusion_signal`. It can also be passed in by user to have a custom margin signal function for whatever classifier they are using. Might need some changes to make it stricter and have solid validation, but that's the general logic I'm thinking to abstract out the margin inclusion signal function to make it model agnostic. Curious to know what your thoughts are.